### DISCUSSION TITLE
Switch to True ES clustering

### DISCUSSION THREAD

**Discussion by 3isenHeiM** [Security-Onion-Solutions/securityonion#9822] [Unsupported Versions]
For the sake of resilience, I'd like to enable the ES true clustering in my distributed architecture. Space is not an issue, since 26To free on each search node and 4To used on each.

I've found some pointers here : https://github.com/Security-Onion-Solutions/securityonion/discussions/4998#discussioncomment-1120022, but I'd like to have more insight as my architecture is in production.

From what I've seen, at some point I would have to re-run the install script on each of the nodes. **Is this really required ?** 

Changing the configuration file on the hosts and restarting the ES service wouldn't do the trick ? From my search this is what running the installer does :

1. Set `true_cluster` to `True` in `/opt/so/saltstack/local/pillar/global.sls`
2. Adding this section in `/opt/so/saltstack/local/pillar/global.sls`
```
elasticsearch:
  config:
    cluster:
      name: <clustername>
```

What about the local changes ? I fear that running the installer on all the nodes will discard my modifications (and put them in an archived folder).

**Comment by cm-ops** [comment]
There is a bit more to converting the cross cluster to true cluster. How many search nodes do you have? Are you looking to retain the Elasticsearch data?

**Reply by 3isenHeiM** [reply]
I have 2 search nodes and I'd like to have the data replicated among them.

And yes I'd like to keep the actual data (shared among the 2 search nodes).

Thanks!

**Comment by cm-ops** [comment]
The manager and two search nodes will all have a unique UUID for their respective cluster. In addition, each search node will have like named indices, for example, `so-zeek-2023.02.21`. Those both need to be addressed. You could clone one search nodes indices to give them a slightly different name, for example, `so-zeek-2023.02.21-clone`, to address the naming. For the UUIDs, you have to start the container, without starting ES and utilize https://www.elastic.co/guide/en/elasticsearch/reference/current/node-tool.html#:~:text=elasticsearch%2Dnode%20detach%2Dcluster%20enables,into%20a%20brand%2Dnew%20cluster. Once joined to the cluster, you would have to import any dangling indices. https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-gateway.html#dangling-indices

Having said the above, this is not something we provide support on, otherwise you would see it in the RTDs. You can make this work, but if things do not work as planned you should be prepared to re-install.
